### PR TITLE
Periodic Matrix Federation testing

### LINKED ISSUE #438: Periodic matrix federation check
> > Also might want to consider doing some type of periodic check on the federation tester 
>
> Was thinking the same! :brain: 

_Originally posted by @noblepayne in https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/issues/381#issuecomment-1237531729_

I'm thinking we can just do a GH action or something to check this on a periodic basis. (Just creating this so we don't forget about it)

### PR DESCRIPTION
**Pull request by CGBassPlayer**
Fixes #438

* Add test for checking `FederationOK` is `true` with [matrix federation tester api](https://federationtester.matrix.org/)
* Add a pytest mark to make tests be able to run periodically
* Add Action to run tests marked as periodic weekly.
* Update E2E action to ignore tests marked as periodic
* Add periodic tests badge to README for quick status reference

### CODE REVIEW

**Comment by CGBassPlayer** [comment]
Yay! Successful tests! 🎉 🎉 🎉

**Review by gerbrent — APPROVED** [review]
Looks good from what I can tell! Will wait on @elreydetoda 's input for final approval. Nice test!

File: test/external/test_matrix.py
Comment (elreydetoda) [inline test/external/test_matrix.py:8]: I'd suggest putting a few more `assert`s in, so if in the future it fails we have a bit more incite as to where it failed (i.e. if it responded with a 500 instead of a 2xx).

I'd also recommend adding comments after the assert, that way it gives us a summary of what the failure is compared to a snippet of the error message. (I need to correct my test comments, I was commenting what they were not what the error would be :sweat_smile:, but I've included what I'll change mine to in yours )

```suggestion
    assert response.ok, "response code wasn't 200-299"
    assert response.body() != "", "response was empty"
    assert response.json()["FederationOK"] == True
```

File: .github/workflows/periodic_test.yml
Comment (elreydetoda) [inline .github/workflows/periodic_test.yml:32]: since you don't have the `if` clause here you can delete this comment.

```suggestion
```

**Review by elreydetoda — CHANGES_REQUESTED** [review]
Overall everything looks awesome @CGBassPlayer!

Had a few comments/suggestions for enhancement of the PR, but all the functionality is there. Nicely done! :partying_face: :rocket::partying_face: :rocket::partying_face: :rocket:

File: .github/workflows/e2e.yml
Comment (CGBassPlayer) [reply .github/workflows/e2e.yml:33]: I didn't even think about that because I was using the local testing to work on the this test 😆 I'll throw that in

Comment (elreydetoda) [reply .github/workflows/e2e.yml:33]: Ya, I honestly don't know either maybe @gerbrent could give some feedback from the JB side if/when they'd like to be notified (in the future)?

Mainly just tossing it out as a question, because I don't know either :grin:

**Comment by CGBassPlayer** [comment]
Last minute addition 😉

**Review by elreydetoda — APPROVED** [review]
Awesome!

LGTM

**Pull request merged**
